apiserver/facades/client/usermanager/usermanager.go

```diff
@@ -369,30 +331,51 @@ func (api *UserManagerAPI) UserInfo(ctx context.Context, request params.UserInfo
 			results.Results = append(results.Results, params.UserInfoResult{Error: apiservererrors.ServerError(apiservererrors.ErrPerm)})
 			continue
 		}
-		if !userTag.IsLocal() {
-			// TODO(wallyworld) record login information about external users.
-			result := params.UserInfoResult{
-				Result: &params.UserInfo{
-					Username: userTag.Id(),
-				},
-			}
-			accessForUser(userTag, &result)
-			results.Results = append(results.Results, result)
-			continue
-		}
 
 		// Get User
-		user, err := api.getLocalUserByTag(ctx, arg.Tag)
+		user, err := api.accessService.GetUserByName(ctx, userTag.Id())
 		if err != nil {
 			results.Results = append(results.Results, params.UserInfoResult{Error: apiservererrors.ServerError(err)})
 			continue
 		}
-		results.Results = append(results.Results, infoForUser(userTag, user))
+		results.Results = append(results.Results, api.infoForUser(ctx, userTag, user))
 	}
 
 	return results, nil
 }
 
+// infoForUser generates a UserInfoResult from a coreuser.User, it fills in
+// information needed but not contained in the core user from the access
+// service.
+func (api *UserManagerAPI) infoForUser(ctx context.Context, tag names.UserTag, user coreuser.User) params.UserInfoResult {
+	result := params.UserInfoResult{
+		Result: &params.UserInfo{
+			Username:       user.Name,
+			DisplayName:    user.DisplayName,
+			CreatedBy:      user.CreatorName,
+			DateCreated:    user.CreatedAt,
+			LastConnection: &user.LastLogin,
+			Disabled:       user.Disabled,
+		},
+	}
+	if user.Disabled {
+		// disabled users have no access to the controller.
+		result.Result.Access = string(permission.NoAccess)
+	} else {
```

Comment:
nit: Always return out early, then you don't need to think about the original if expression, then you don't need to nest in the else expression.

```suggestion
	if user.Disabled {
		// disabled users have no access to the controller.
		result.Result.Access = string(permission.NoAccess)
	    return result
	}
```

Reply:
Good point, changed